fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraBaseResource.java

```diff
@@ -94,4 +95,22 @@ protected void setUpJMSInfo(final UriInfo uriInfo, final HttpHeaders headers) {
         }
     }
 
+    /**
+     * Produce a baseURL for JMS events using the system property fcrepo.jms.baseUrl of the form host[:port], if it
+     * exists.
+     * 
+     * @return String the base Url
+     */
+    protected String getBaseUrlProperty() {
+        final String propBaseURL = System.getProperty("fcrepo.jms.baseUrl", "");
+        if (propBaseURL.length() > 0) {
+            if (propBaseURL.indexOf(':') > -1) {
+                final String[] parts = propBaseURL.split(":");
+                return uriInfo.getBaseUriBuilder().clone().host(parts[0]).port(Integer.valueOf(parts[1])).toString();
```

Comment:
Before I do this, is it your expectation that if I am running Fcrepo at "http://blah.com:8080" I might want the JMS messages to reference "https://blah.com:8443"? I couldn't think of why you would, but now I'm thinking it might be easier to pass the whole property to UriBuilder.uri(String s)?